backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -44,24 +48,32 @@
   private static final String JSON_SECOND_PARTICIPANT_USERNAME = "secondParticipantUsername";
   private static final String JSON_DURATION = "duration";
 
+  // Create GmailFactory and use it to instantiate Email Notifier
+  private final GmailFactory gmFactory = new GmailFactory();
+  private final EmailNotifier emailNotifier = new EmailNotifier(gmFactory.build());
+
+  // Get DatastoreService and instiate Match and Participant Datastores
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final MatchDatastore matchDatastore = new MatchDatastore(datastore);
+  private final ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+
+  // Get participant username
+  UserService userService = UserServiceFactory.getUserService();
+
+  public SearchMatchServlet() throws GeneralSecurityException, IOException {}
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
     System.out.println("Request received");
 
-    // Get participant username
-    UserService userService = UserServiceFactory.getUserService();
     String email = userService.getCurrentUser().getEmail();
     if (email == null) {
       response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid email.");
       return;
     }
     String username = email.split("@")[0];
```

Comment:
WDYT about potentially extracting this out into its own method, so that the doGet becomes a bit clearer to read?

Reply:
I am confused as to what you mean in this case.

Reply:
@dhruvi618 i know what you mean, I can do this part

Reply:
i have extracted a get username that uses this logic If that is what you mean ?